### PR TITLE
Draft: Wagtail 4.2 support

### PR DESCRIPTION
**Pull request by katdom13**
- Updated tox testing and workflow actions
- Removed support for Wagtail versions < 4.1
- Suppress warnings
- Code cleanup

All tests run OK

```
Found 19 test(s).
Creating test database for alias 'default'...
System check identified no issues (0 silenced).
...................
----------------------------------------------------------------------
Ran 19 tests in 0.920s

OK
```

### CODE REVIEW

**Review by chosak — COMMENTED** [review]
Thank you @katdom13! We don't typically make packages releases for Wagtail release candidates but can definitely do this once 4.2 is out.

Along with this change we could probably also remove 4.0 from the tox/GitHub testing matrix - we should keep 4.1 as it's an LTS but explicitly testing 4.0 doesn't seem necessary. Any thoughts @willbarton?

**Comment by katdom13** [comment]
Hi @chosak and @willbarton ,
Would you also like me to drop support for Wagtail < 4.1 altogether? This would also mean removing code inside conditionals (e.g. inside `if WAGTAIL_VERSION >= (x, y)`) according to [the guidelines]?(https://github.com/wagtail/wagtail/wiki/Python-Package-Maintenance-Guidelines#setuppy--setupcfg--pyprojecttoml) 
Please note the EOL and support: https://endoflife.date/wagtail

Thank you!

**Comment by katdom13** [comment]
Hi @chosak , @willbarton 
This MR should drop versions of Wagtail < 4.1.
Thanks!

File: wagtailinventory/tests/test_helpers.py [hunk 1]
```diff
@@ -12,7 +12,7 @@
 
 
 if wagtail.VERSION < (3, 0):  # pragma: nocover
```
Comment (nickmoreton) [inline wagtailinventory/tests/test_helpers.py:14]: Could this be removed and below you can then not need to use the `CORE_BLOCKS` and use the full string.

e.g. `wagtail.blocks.field_block.CharBlock`

File: wagtailinventory/tests/test_wagtail_hooks.py [hunk 2]
```diff
@@ -2,14 +2,14 @@
...
 import wagtail
-from wagtail.core.models import Page, Site
-from wagtail.tests.utils import WagtailTestUtils
+from wagtail.models import Page, Site
+from wagtail.test.utils import WagtailTestUtils
 
 from wagtailinventory.models import PageBlock
 
 
 if wagtail.VERSION < (3, 0):  # pragma: nocover
```
Comment (nickmoreton) [inline wagtailinventory/tests/test_wagtail_hooks.py:11]: Ditto

**Review by nickmoreton — COMMENTED** [review]
LGTM. Just a couple of possible changes

File: wagtailinventory/tests/test_helpers.py (hunk 1 above)
Comment (katdom13) [reply wagtailinventory/tests/test_helpers.py]: Hi @nickmoreton ,
Good catch! This has been updated in the latest commit https://github.com/cfpb/wagtail-inventory/pull/66/commits/b2e4161501da12d19d4cafe3bb148b301e31c667
This should also resolve the one below.

**Review by nickmoreton — APPROVED** [review]
LGTM 👍

**Comment by willbarton** [comment]
@katdom13 for us, it would be preferable to maintain Wagtail 2.15 support a bit longer, as that's where we currently are (but are in the process of updating).

Our preference is to test against LTS releases, so that would be 2.15 and 4.1. Unfortunately that means also maintaining the `if WAGTAIL_VERSION >= (x, y)` tests for now.

**Comment by nickmoreton** [comment]
> @katdom13 for us, it would be preferable to maintain Wagtail 2.15 support a bit longer, as that's where we currently are (but are in the process of updating).
> 
> Our preference is to test against LTS releases, so that would be 2.15 and 4.1. Unfortunately that means also maintaining the `if WAGTAIL_VERSION >= (x, y)` tests for now.

Thanks @willbarton 

@katdom13 If you will have time to look at this that would be great but let me know if you would like me to pick it up? Thanks

**Comment by willbarton** [comment]
@katdom13 @nickmoreton I can pick this up. We're in the midst of converting our projects to `pyproject.toml` too, so I'm going to pick up the commits here, do that conversion and we'll do a release based on that.

Thanks for your patience!

**Comment by willbarton** [comment]
Thank you so much @katdom13 and @nickmoreton! These changes have been merged in #67 (fascinating that GitHub sees that and indicates this PR is merged now too!)

We'll get a release out with them shortly!

**Pull request merged**